src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForFailedStatus.java

```diff
@@ -192,9 +193,22 @@ public void should_record_generic_failure_when_metadata_parsing_fails() throws E
 
     @Test
     public void should_record_generic_failure_when_zip_contains_documents_not_in_pdf_format() throws Exception {
+        checkFailureEvent("5_24-06-2018-00-00-00.zip", Event.DOC_FAILURE);
+    }
+
+    @Test
+    public void should_record_signature_failure_when_zip_contains_invalid_signature() throws Exception {
+        processor.signatureAlg = "sha256withrsa";
+        processor.publicKeyBase64 =
+            "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDEXfjyDFFigzsmFvTe2cWZ45ggH/XoS/3C6Ur/"
+            + "V0egi8k5hnIIgPEOUqhrX5UcQorSX7bIlMped6TtPkYdGs/QI6S5m2uz+6Mjai7ZfACGhYxIs8"
+            + "35msqvRsDM0tIle/h3eZJb7iPE0anMWb8MkBYU3D3vAnPdBZxiEIwNMUNzqQIDAQAB";
+        checkFailureEvent("43_24-06-2018-00-00-00.test.zip", Event.DOC_SIGNATURE_FAILURE);
+    }
+
+    public void checkFailureEvent(String invalidZipFile, Event event) throws Exception {
         // given
-        String noPdfZip = "5_24-06-2018-00-00-00.zip";
```

Comment:
At least we knew what the problem was originally.
A comment on line 196 could help.

Reply:
The method name is quite clear on what the problem is there. I am not sure I understand this comment. What comment would you add?